rspace/src/test/scala/coop/rchain/rspace/merging/MergingLogic.scala

```diff
@@ -0,0 +1,48 @@
+package coop.rchain.rspace.merging
+
+import coop.rchain.rspace.merger.MergingLogic._
+import org.scalatest.{FlatSpec, Matchers}
+
+class MergingLogic extends FlatSpec with Matchers {
+  "1" should "" in {
+    val conflictMap = Map(
+      1 -> Set(2, 3, 4),
+      2 -> Set(1),
+      3 -> Set(1, 2),
+      4 -> Set(1)
+    )
+    val rjOptions = computeRejectionOptions(conflictMap)
+    rjOptions shouldBe Set(Set(1), Set(1, 2), Set(2, 3, 4))
+  }
+
+  "2" should "" in {
+    val conflictMap = Map(
+      1 -> Set(2, 3, 4),
+      2 -> Set(1, 3, 4),
+      3 -> Set(1, 2, 4),
+      4 -> Set(1, 2, 3)
+    )
+    val rjOptions = computeRejectionOptions(conflictMap)
+    rjOptions shouldBe Set(Set(2, 3, 4), Set(1, 3, 4), Set(1, 2, 4), Set(1, 2, 3))
+  }
+
+  "3" should "" in {
+    val conflictMap = Map(
+      1 -> Set(2, 3, 4),
+      2 -> Set(1),
+      3 -> Set(1, 4),
+      4 -> Set(1, 3)
+    )
+    val rjOptions = computeRejectionOptions(conflictMap)
+    rjOptions shouldBe Set(Set(2, 3, 4), Set(1, 3), Set(1, 4))
```

Comment:
Previous algorithm computing rejection options was incomplete, e.g. this test were not passing. I don't quite like how these tests are organised: it's not the whole merging logic and names for tests do not tell much, maybe I'll come up with something better.